lib/libkafka_asio/detail/impl/request_write.h

```diff
@@ -80,7 +80,13 @@ inline void WriteBytes(const Bytes& value, std::ostream& os)
     return;
   }
   WriteInt32(static_cast<Int32>(value->size()), os);
+#if __cplusplus >= 201103L
   os.write( reinterpret_cast<const char*>( value->data() ), value->size() );
+#else
+  for (int i=0; i<value->size(); ++i) {
+    os.put((*value)[i]);
```

Comment:
could this also be replaced with?:
```cpp
if (!value->empty())
{
  os.write(reinterpret_cast<const char*>(&(*value)[0]), value->size());
}
```
(Didn't try to compiled it...)

Reply:
The C++ standard suggests it should work. I'll try it!

http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2013/n3690.pdf, Section 23.3.7.1